compiler/il/OMRNode.cpp

```diff
@@ -7727,3 +7727,27 @@ OMR::Node::isEAEscapeHelperCall()
 
    return false;
    }
+
+
+TR::Node *
+OMR::Node::storeToAddressField(TR::Node *obj, TR::SymbolReference *symRef, TR::Node *value)
+   {
+   TR::Compilation *comp = TR::comp();
+   TR::Node * node;
+
+   if (TR::Compiler->om.writeBarrierType() != gc_modron_wrtbar_none)
+      {
+      node = TR::Node::createWithSymRef(TR::awrtbari, 3, 3, obj, value, obj, symRef);
+      }
+   else
+      {
+      node = TR::Node::createWithSymRef(TR::astorei, 2, 2, obj, value, symRef);
+      }
```

Comment:
If we have a method like this for indirect stores, should we have one for direct stores stores as well?

Reply:
The idea of this method was to help with various write barrier checks. Direct store is much simpler. I think we can create a helper for it if a need arises.